src/index.ts

```diff
@@ -1,162 +1,625 @@
+/*
+
+    %@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@%
+  @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+%@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@%
+@@@@@%     %@@@@@@%         %@@@@@@@%     %@@@@@
+@@@@@       @@@@@%            @@@@@@       @@@@@
+@@@@@%      @@@@@             %@@@@@      %@@@@@
+@@@@@@%     @@@@@     %@@%     @@@@@     %@@@@@@
+@@@@@@@     @@@@@    %@@@@%    @@@@@     @@@@@@@
+@@@@@@@     @@@@%    @@@@@@    @@@@@     @@@@@@@
+@@@@@@@    %@@@@     @@@@@@    @@@@@%    @@@@@@@
+@@@@@@@    @@@@@     @@@@@@    %@@@@@    @@@@@@@
+@@@@@@@    @@@@@@@@@@@@@@@@     @@@@@    @@@@@@@
+@@@@@@@    %@@@@@@@@@@@@@@@     @@@@%    @@@@@@@
+@@@@@@@     %@@%     @@@@@@     %@@%     @@@@@@@
+@@@@@@@              @@@@@@              @@@@@@@
+@@@@@@@%            %@@@@@@%            %@@@@@@@
+@@@@@@@@@%        %@@@@@@@@@@%        %@@@@@@@@@
+%@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@%
+  @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+    %@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@%
+
+ */
```

Comment:
Most important file, everything comes together here. The documentation should be self-explanatory, note if anything is not clear or missing.